File: 8.Ros/Ros2/netpi/src/subscriber.cpp
/*
    export ROS_DOMAIN_ID=30
    export ROS_IP=192.168.1.30
    export ROS_MASTER_URI=http://192.168.1.30:11311
*/



#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"
#include <chrono>
#include <iostream>

using namespace std::chrono_literals;

int main()
{

    rclcpp::init(0, nullptr);
    auto node = rclcpp::Node::make_shared("subscriber");
    auto pub = node->create_subscription<std_msgs::msg::String>("chatter", 10, 
        [&](const std_msgs::msg::String::SharedPtr msg) {
            RCLCPP_INFO(node->get_logger(), "I heard: '%s'", msg->data.c_str());
        });
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: 8.Ros/Ros2/soso/soso/subscriber.py
import rclpy
from std_msgs.msg import String

def main():
    rclpy.init()
    node = rclpy.create_node('soso_subscriber')
    sub = node.create_subscription(String, '/topic', lambda msg: node.get_logger().info('I heard: "%s"' % msg.data), 10)
    rclpy.spin_once(node)
    node.destroy_node()
    rclpy.shutdown()


if __name__ == '__main__':
    main()